File: task_equivalents/llm_translations/starcoder/base/r/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.R
library(tidyverse)
library(e1071) # for SVMs

Auto <- read.csv("data/csv/Auto.csv", na.strings="?") %>%
  drop_na() %>%
  mutate(mpg01 = ifelse(mpg > median(mpg), 1, 0)) %>%
  select(-mpg, -name)
var_no_scale <- c("cylinders", "year", "origin", "mpg01")
var_to_scale <- c("displacement", "horsepower", "weight", "acceleration")
scaled_var <- scale(Auto[, var_to_scale])
df <- cbind(scaled_var, Auto[, var_no_scale])
x <- df[, -ncol(df)]
y <- df[, ncol(df)]
tune_param <- list(C = c(0.01, 0.1, 1, 10, 100, 1000))
svm_linear <- tune(svm, train.x=x, train.y=as.factor(y), kernel="linear", ranges=tune_param,
                   cross=5, scale=FALSE)
tune_param <- list(C = c(0.01, 0.1, 1, 10, 100, 1000), degree = 2:5)
svm_poly <- tune(svm, train.x=x, train.y=as.factor(y), kernel="polynomial", ranges=tune_param,
                 cross=5, scale=FALSE)
tune_param <- list(C = c(0.01, 0.1, 1, 10, 100, 1000), gamma = c(0.5, 1, 2, 3))
svm_rbf <- tune(svm, train.x=x, train.y=as.factor(y), kernel="radial", ranges=tune_param,
                cross=5, scale=FALSE)

File: task_equivalents/llm_translations/starcoder/base/python/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.py
import pandas as pd
from sklearn.preprocessing import scale
from sklearn.svm import SVC
from sklearn.model_selection import GridSearchCV

df = pd.read_csv('data/csv/Auto.csv', na_values='?', index_col=0)
df = df.dropna()

df['mpg01'] = (df['mpg'] > df['mpg'].median()).astype(int)
df = df.drop(['mpg', 'name'], axis=1)

var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']
scaled_var = scale(df[var_to_scale])
df = pd.concat([pd.DataFrame(scaled_var, columns=var_to_scale), df[var_no_scale]], axis=1)

x = df.drop('mpg01', axis=1)
y = df['mpg01']
ctry = [0.01, 0.1, 1, 10, 100, 1000]

param_grid = {'C': ctry}
svm_linear = SVC(kernel='linear')
grid_search = GridSearchCV(svm_linear, param_grid, cv=5)
grid_search.fit(x, y)
print(grid_search.best_params_)
print(grid_search.best_score_)

polytry = [2, 3, 4, 5]
param_grid = {'C': ctry, 'degree': polytry}
svm_poly = SVC(kernel='poly')
grid_search = GridSearchCV(svm_poly, param_grid, cv=5)
grid_search.fit(x, y)
print(grid_search.best_params_)
print(grid_search.best_score_)

gammatry = [0.5, 1, 2, 3]
param_grid = {'C': ctry, 'gamma': gammatry}
svm_rbf = SVC(kernel='rbf')
grid_search = GridSearchCV(svm_rbf, param_grid, cv=5)
grid_search.fit(x, y)
print(grid_search.best_params_)
print(grid_search.best_score_)